duckduckgo/macos-browser
# Prevents launching our menu agent without an auth code.

Task/Issue URL: https://app.asana.com/0/0/1205112155849085/f
BSK PR: https://github.com/duckduckgo/BrowserServicesKit/pull/469
iOS PR: https://github.com/duckduckgo/iOS/pull/1928

## Description:

Prevents launching our menu agent app without a valid NetP auth code.

## Steps to test this PR:

Before running these tests, open Keychain.app and remove all auth tokens.  You'll need to enable NetP again through an invite code.

Start in `develop`

1. Run NetP normally using the latest from `develop` (this is important so the auth token is not shared).
2. Connect.
3. Once connected, reset the system extension and login item (through the debug menu).
4. Quit the app.
5. Open Keychain.app, search for the auth token that matches your app build type and make sure it shows the following access group:

<img width="336" alt="Screenshot 2023-08-21 at 21 22 10" src="https://github.com/duckduckgo/macos-browser/assets/1836005/8f58dbf9-6e48-4b2c-a432-c1da0e6ee56a">

6. Switch to this branch
7. Start the app
8. Open Keychain.app, search for the app token, it should have been moved to a build specific app group:

<img width="623" alt="Screenshot 2023-08-21 at 21 23 01" src="https://github.com/duckduckgo/macos-browser/assets/1836005/694e6f38-bfe6-474d-b0e4-da6c5d5a9401">

9. Start NetP and make sure it works.
10. Stop NetP, reset extension and login item through the debug menu
11. Open Keychain.app, and remove all auth tokens.
12. Go to /Applications/DEBUG/DuckDuckGo.app/Contents/Library/LoginItems/
13. Double click on `DuckDuckGo Agent`
14. Make sure the menu app does not open for you - ie: you can't see the icon in the status bar.


---
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
[Pull Request Documentation](https://app.asana.com/0/1202500774821704/1204012835277482/f)
